xblock/fields.py

```diff
@@ -549,7 +558,7 @@ def __delete__(self, xblock):
         # Since we know that the field_data no longer contains the value, we can
         # avoid the possible database lookup that a future get() call would
         # entail by setting the cached value now to its default value.
-        self._set_cached_value(xblock, copy.deepcopy(self.default))
+        self._set_cached_value(xblock, self._get_default_value_to_cache(xblock))
```

Comment:
@cpennington I was seeing a failure in a test that had a String field with a default of `UNIQUE_ID` when the field was being set and then deleted. This change fixed the failure - but is it correct?